### PR TITLE
update SolverCore dep

### PR DESCRIPTION
**Pull request by dpo**
prepare for new solver behavior

### CODE REVIEW

**Comment by dpo** [comment]
After this PR, all solvers should be updated to conform to the new API and behavior.

**Comment by tmigot** [comment]
@dpo  this is ready to be checked

File: src/tron.jl [hunk 1]
```diff
@@ -265,7 +265,8 @@ function solve!(
     status = :unbounded
   end
 
-  stats = GenericExecutionStats(status, nlp)
+  stats = GenericExecutionStats(nlp)
+  set_status!(stats, status)
```
Comment (dpo) [inline src/tron.jl:269]: Same here.

**Review by dpo — COMMENTED** [review]
Thank you for picking this up. Should we also move `stats` outside the solver?

File: src/lbfgs.jl [hunk 2]
```diff
@@ -188,7 +188,8 @@ function solve!(
     end
   end
 
-  stats = GenericExecutionStats(status, nlp)
+  stats = GenericExecutionStats(nlp)
+  set_status!(stats, status)
```
[the remark this answers is not in the document]
Comment (tmigot) [reply src/lbfgs.jl:192]: What do you mean by is not updated? I add it to the stats here https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/db98717db45c4ae2482462a85920e17396925bc6/src/lbfgs.jl#L196 and `iter` is updated there https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/db98717db45c4ae2482462a85920e17396925bc6/src/lbfgs.jl#L173

**Comment by tmigot** [comment]
> Thank you for picking this up. Should we also move `stats` outside the solver?

I think a follow-up PR could had the function `solve!(solver, nlp, stats)`

File: src/lbfgs.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (dpo) [reply src/lbfgs.jl:192]: I was thinking of the callback, but this version of lbfgs doesn't yet use the callback. Should we add it later?

[the remark this answers is not in the document]
Comment (tmigot) [reply src/lbfgs.jl:192]: Yes, it is a good idea.

**Review by dpo — COMMENTED** [review]
I can't approve my own PR. Approved.

**Pull request merged**